### PR TITLE
[SYCL][ESIMD] Move rdtsc function out of experimental namespace

### PR DESCRIPTION
**Pull request by fineg74**

### CODE REVIEW

File: sycl/include/sycl/ext/intel/esimd/math.hpp [hunk 1]
```diff
@@ -101,10 +101,10 @@ __esimd_abs_common_internal(simd<TArg, SZ> src0) {
 }
 
 template <typename TRes, typename TArg>
-ESIMD_NODEBUG ESIMD_INLINE
-    std::enable_if_t<detail::is_esimd_scalar<TRes>::value &&
-                         detail::is_esimd_scalar<TArg>::value,
-                     TRes>
+ESIMD_NODEBUG
+    ESIMD_INLINE std::enable_if_t<detail::is_esimd_scalar<TRes>::value &&
```
Comment (sarnex) [inline sycl/include/sycl/ext/intel/esimd/math.hpp:105]: nit: there are some whitespace changes of this pattern that in my opinion decrease readability, i would expect we only need to touch the rdtsc function and not these

File: sycl/include/sycl/ext/intel/esimd/math.hpp [hunk 2]
```diff
@@ -1302,6 +1301,13 @@ __ESIMD_API uint32_t subb(uint32_t &borrow, uint32_t src0, uint32_t src1) {
   return Res[0];
 }
 
+/// rdtsc - get the value of timestamp counter.
+/// \return the current value of timestamp counter
```
Comment (sarnex) [inline sycl/include/sycl/ext/intel/esimd/math.hpp:1305]: I think it's `@return` based on other functions

File: sycl/test-e2e/ESIMD/rdtsc.cpp [hunk 3]
```diff
@@ -41,9 +41,9 @@ int test_rdtsc() {
       auto Kernel = ([=](sycl::nd_item<1> ndi) [[intel::sycl_explicit_simd]] {
         using namespace sycl::ext::intel::esimd;
         auto Idx = ndi.get_global_id(0);
-        uint64_t StartCounter = sycl::ext::intel::experimental::esimd::rdtsc();
```
Comment (sarnex) [inline sycl/test-e2e/ESIMD/rdtsc.cpp:44]: nit: maybe we call both experimental and non-experimental based on `Idx % 2` or something

**Review by v-klochkov — COMMENTED** [review]
Looks good to me.
Can you please check if this [extension](http://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/KHR/SPV_KHR_shader_clock.html) is ready for usage, maybe create a copy/draft of this PR with OpReadClockKHR

**Review by v-klochkov — APPROVED** [review]
The change is correct as written now, and the experiment with using SPIRV instead of GenX intrinsic can be done in a separate PR.

**Pull request merged**